pkg/upload/android-aab.go

```diff
@@ -2,138 +2,136 @@ package upload
 
-		if !utils.FileExists(config["aabPath"]) {
-			log.Info(variant + " does not have a valid .aab file")
-			continue
+			if err != nil {
+				return fmt.Errorf("error reading raw AAB manifest data. " + err.Error())
+			}
+		} else {
+			return fmt.Errorf("unable to read data from " + aabManifestPath + " " + err.Error())
 		}
 
-		outputPath := filepath.Join(strings.Replace(config["aabPath"], filepath.Base(config["aabPath"]), "", -1), "raw")
-
-		log.Info("Expanding " + filepath.Base(config["aabPath"]))
-
-		err := utils.Unzip(config["aabPath"], outputPath)
-
 		if err != nil {
 			return err
 		}
 
-		log.Success(filepath.Base(config["aabPath"]) + " expanded")
-
-		log.Info("Reading data from AndroidManifest.xml")
-
-		aabManifestData, err := android.ReadAabManifest(filepath.Join(outputPath, "base", "manifest", "AndroidManifest.xml"))
+		if apiKey == "" {
+			apiKey = manifestData["apiKey"]
+			log.Info("Using " + apiKey + " as API key from AndroidManifest.xml")
+		}
 
-		if err != nil {
-			return fmt.Errorf("error reading raw AAB manifest data. " + err.Error())
+		if applicationId == "" {
+			applicationId = manifestData["applicationId"]
+			log.Info("Using " + applicationId + " as application ID from AndroidManifest.xml")
 		}
 
-		if getApiKeyFromManifest {
-			apiKey = aabManifestData["apiKey"]
+		if buildUuid == "" {
+			buildUuid = manifestData["buildUuid"]
+			log.Info("Using " + buildUuid + " as build UUID from AndroidManifest.xml")
 		}
 
-		err = android.ProcessProguard(apiKey, variant, outputPath, aabManifestData, overwrite, timeout, numberOfVariants, endpoint, failOnUploadError)
+		if versionCode == "" {
+			versionCode = manifestData["versionCode"]
+			log.Info("Using " + versionCode + " as version code from AndroidManifest.xml")
+		}
 
-		if err != nil {
-			return fmt.Errorf("error processing Proguard mapping file. " + err.Error())
+		if versionName == "" {
+			versionName = manifestData["versionName"]
+			log.Info("Using " + versionName + " as version name from AndroidManifest.xml")
 		}
+	}
 
-		androidNdkRoot, err := android.GetAndroidNDKRoot("")
+	soFilePath := filepath.Join(tempDir, "BUNDLE-METADATA", "com.android.tools.build.debugsymbols")
+
+	if utils.FileExists(soFilePath) {
+		fileList, err := utils.BuildFileList([]string{soFilePath})
 
 		if err != nil {
-			return err
+			return fmt.Errorf("error building `.so` file list. " + err.Error())
 		}
 
-		log.Info("Using Android NDK located here: " + androidNdkRoot)
+		for _, file := range fileList {
+			err = ProcessAndroidNDK(apiKey, applicationId, androidNdkRoot, "", []string{file}, projectRoot, "", versionCode, versionName, endpoint+"/ndk-symbol", failOnUploadError, retries, timeout, overwrite, dryRun)
 
-		log.Info("Locating objcopy within Android NDK path")
-
-		objCopyPath, err := android.BuildObjcopyPath(androidNdkRoot)
-
-		if err != nil {
-			return err
+			if err != nil {
+				return err
+			}
 		}
+	} else {
+		log.Info("Skipping NDK processing")
+	}
```

Comment:
If you have no .so files in your AAB (presumably as you have no native code), what does the directory structure look like? Do you have no `com.android.tools.build.debugsymbols` or is the directory empty?

We want to make sure we don't error in the case of there being no NDK files.

I think the Info message is good, but maybe word it in a way that makes it obvious that it's not a problem:
```suggestion
	} else {
		log.Info("No NDK (.so) files detected for upload.")
	}
```